### PR TITLE
Feature/Use authority as a field

### PR DESCRIPTION
**Pull request by friedger**
:x: 
This PR is to visualise the impact of not using Reflector but defining the authority from the context.

### CODE REVIEW

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java [hunk 1]
```diff
@@ -479,12 +483,12 @@ public boolean isOnCache() {
                 || mDestination == Downloads.Impl.DESTINATION_CACHE_PARTITION_PURGEABLE);
     }
 
-    public Uri getMyDownloadsUri() {
-        return ContentUris.withAppendedId(Downloads.Impl.CONTENT_URI, mId);
+    public Uri getMyDownloadsUri(String authority) {
```
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java]: this is publicly accessible, we don't want the client to have to pass their authority. Either we make this method package? or we do somethign else

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java [hunk 2]
```diff
@@ -118,27 +119,32 @@
...
-        sURIMatcher.addURI(AUTHORITY, "all_downloads", ALL_DOWNLOADS);
-        sURIMatcher.addURI(AUTHORITY, "all_downloads/#", ALL_DOWNLOADS_ID);
-        sURIMatcher.addURI(AUTHORITY, "my_downloads/#/" + Downloads.Impl.RequestHeaders.URI_SEGMENT, REQUEST_HEADERS_URI);
-        sURIMatcher.addURI(AUTHORITY, "all_downloads/#/" + Downloads.Impl.RequestHeaders.URI_SEGMENT, REQUEST_HEADERS_URI);
-        // temporary, for backwards compatibility
-        sURIMatcher.addURI(AUTHORITY, "download", MY_DOWNLOADS);
-        sURIMatcher.addURI(AUTHORITY, "download/#", MY_DOWNLOADS_ID);
-        sURIMatcher.addURI(AUTHORITY, "download/#/" + Downloads.Impl.RequestHeaders.URI_SEGMENT, REQUEST_HEADERS_URI);
-        sURIMatcher.addURI(AUTHORITY, Downloads.Impl.PUBLICLY_ACCESSIBLE_DOWNLOADS_URI_SEGMENT + "/#", PUBLIC_DOWNLOAD_ID);
+    public static void initURIMatcher(Context context) {
```
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java]: `private`

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java [hunk 3]
```diff
@@ -118,27 +119,32 @@
...
+                    Downloads.Impl.ALL_DOWNLOADS_CONTENT_URI(AUTHORITY),
+            };
+
+            sURIMatcher.addURI(AUTHORITY, "my_downloads", MY_DOWNLOADS);
+            sURIMatcher.addURI(AUTHORITY, "my_downloads/#", MY_DOWNLOADS_ID);
+            sURIMatcher.addURI(AUTHORITY, "all_downloads", ALL_DOWNLOADS);
+            sURIMatcher.addURI(AUTHORITY, "all_downloads/#", ALL_DOWNLOADS_ID);
+            sURIMatcher.addURI(AUTHORITY, "my_downloads/#/" + Downloads.Impl.RequestHeaders.URI_SEGMENT, REQUEST_HEADERS_URI);
+            sURIMatcher.addURI(AUTHORITY, "all_downloads/#/" + Downloads.Impl.RequestHeaders.URI_SEGMENT, REQUEST_HEADERS_URI);
+            // temporary, for backwards compatibility
```
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java]: lets delete this stuff? @amlcurran ?

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadScanner.java [hunk 4]
```diff
@@ -39,6 +39,7 @@
 
     private final Context mContext;
     private final MediaScannerConnection mConnection;
+    private final String mAuthority;
```
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadScanner.java]: lets use Novoda style naming so we know what we've changed/added

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java (hunk 3 above)
Comment (amlcurran) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java]: sure, so long as those URIs aren't declared anywhere

File: demo/build.gradle [hunk 5]
```diff
@@ -19,7 +19,7 @@ android {
         versionCode 1
         versionName "1.0"
 
-        manifestPlaceholders = [downloadAuthority: "${applicationId}"]
+        manifestPlaceholders = [downloadAuthority: "${applicationId}.downloads"]
```
Comment (blundell) [inline demo/build.gradle]: this and https://github.com/novoda/download-manager/pull/13/files#diff-d18483ff140c15dd49ed3d88822df403R205 , 1 is the client and 1 is the library therefore the connection is not apparent. Here is lies a problem

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java (hunk 3 above)
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java]: yeah delete all that it uses I mean - simplify (as we do not have / need backwards compat)

File: demo/build.gradle (hunk 5 above)
Comment (friedger) [inline demo/build.gradle]: how the authority is determined can be more sophisticated. I think convention is still the best way to go.

Comment (blundell) [inline demo/build.gradle]: We want the client to either completely control authority creation, or be completely agnostic of it (and we just use context). Not in the middle

**Comment by blundell** [comment]
unmergable, but considered

**Pull request closed without merging**